build.project

```diff
@@ -30,21 +23,39 @@ if [ -z "$TARGET" ]; then
     exit 1
 fi
 
+
 TARGET_FAMILY=$(echo "$TARGET" | awk -F- '{ print $1 }')
-TARGET_SUBFAMILY="$(echo "$TARGET" | awk -F- '{ print $1"-"$2 }')"
+TARGET_SUBFAMILY="$(echo "$TARGET" | awk -F- '{ print $2 }')"
+TARGET_PROJECT="$(echo "$TARGET" | awk -F- '{ print $3 }')"
+
+
+BUILD_TARGET_FILE="_targets/build.project.${TARGET_FAMILY}-${TARGET_SUBFAMILY}"
+BUILD_PROJECT_FILE="_projects/build.project.${TARGET_FAMILY}-${TARGET_SUBFAMILY}-${TARGET_PROJECT}"
+
 
-BUILD_TARGET_FILE="_targets/build.project.${TARGET_SUBFAMILY}"
 if [ ! -f "$BUILD_TARGET_FILE" ]; then
     b_list_available_targets "TARGET='$TARGET' is unavailable for this project"
     exit 1
 fi
 
 . "$BUILD_TARGET_FILE"
 
+
```

Comment:
Non-intuitive logic flow, I would write:
```bash
if [ -n "$TARGET_PROJECT" ]; then
    if [ ! -f "$BUILD_PROJECT_FILE" ]; then
        b_list_available_targets "TARGET='$TARGET' is unavailable for this project"
        exit 1
    fi

    . "$BUILD_PROJECT_FILE"
fi
```

The main logic is to include the file **if we have project defined**, not if the file exists.

I'm not even sure why checking the last char of `$TARGET` variable is even needed in Your code.

Reply:
Ok, the logic flow can be changed.

Checking the last character covers the case when user defines `TARGET` as follow:
`armv7m7-imxrt106x-`
Then:
- `$TARGET_PROJECT`  is empty 
- `$TARGET` contains wrong char at the end
- in your proposition this mistake will not be caught

Reply:
Ok, You're talking about setting for example `TARGET="ia32-generic-"`.

This is not a problem related to the existence of `$BUILD_PROJECT_FILE`, so it should not be checked there. 

Currently, having `-` at the end of `TARGET` won't break anything, but we can check it nevertheless. The other approach would be to always translate `TARGET` into triplet (eg. `ia32-generic` => `ia32-generic-none`) - that way it's handled in the autotools/automake (eg. when we compile `gcc`, the `arm-phoenix` target is translated to `arm-none-phoenix`).

I would stick with checking format of `TARGET` variable if You feel it's needed - but checking just the suffix might not be enough (eg. it won't catch `ia32-generic--project` as an error). Maybe we can check the whole format:

```bash
if [ "$TARGET" != "$TARGET_FAMILY-$TARGET_SUBFAMILY" ] || [ "$TARGET" != "$TARGET_FAMILY-$TARGET_SUBFAMILY-$TARGET_PROJECT" ]; then
[...]
```
but probably it would be better to rewrite the TARGET_FAMILY/SUBFAMILY/PROJECT  assignment to use switch-case with patterns and default case with error message.

Reply:
I've made changes as you proposed associated with checking the whole format. I think this verification is needed. Some projects has hardcode `TARGET` name as directory name.

Reply:
ok, fine be me.